### ISSUE TITLE
When I try to delete inline pictures that I have added (via the addin), Word doesn't delete them all.

### ISSUE THREAD

**Issue by wbecker** [OfficeDev/office-js#2794]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

* Operating System: Office on web
* Browser (if using Office on the web): Firefox

## Expected behavior
I am adding images under paragraphs to alert the user of problems. I may sometimes add several images under each paragraph. I add a altTextDescription to these images so that I can keep track of them and remove them. I then delete them:

```
removeButtonsFromDocument() {
    window.Word.run(
      async context => {
        const pictures = context.document.body.inlinePictures;
        pictures.load("altTextDescription");
        await context.sync();

        const picturesToDelete = pictures.items.filter(picture => picture.altTextDescription.startsWith(MY_PREFIX));

        if (picturesToDelete.length) {
          console.log("Deleting", picturesToDelete.length, "items");

          for (let index = 0; index < picturesToDelete.length; index += 1) {
            console.log("delete pic", index);
            const picture = picturesToDelete[index];
            picture.delete();
            await context.sync();
          }
        }
      }
    );
  }
```

I expect this code to delete all the images with my prefix (I have tried moving the context.sync out of the loop - it has no effect)

## Current behavior
I am finding 2 problems:
1. If I call this function mulitple times, `context.document.body.inlinePictures` still has the same amount pictures. Calling picture.delete doesn't seem to remove it from the list (even if I ask for a fresh context and re-load the pictures
2. Not all the pictures are deleted. It seems to delete only 1 item per paragraph. If I call the function multiple times it will remove them all. However, because of the first issue, I can't use `context.document.body.inlinePictures` to detect how many pictures are left (because it doesn't decrease), and can't use this as a reference. 

However if I reload the addin, it does seem to update the amount of inlinePictures detected in `context.document.body.inlinePictures`.

## Steps to reproduce
1. Add some 3 images to 3 paragraphs (9 in total) - using paragraph.insertInlinePictureFromBase64(). Set a unique altTextDescription so you don't delete other existing images.
2. Try to delete the images using the above code.
3. Note images remain.

Here is a log of the deletion process: I think the `[RumShepherd]` log is associated with a delete actually occurring. When no images are left, it doesn't appear.
![image](https://user-images.githubusercontent.com/348044/186458157-d85e0f9b-3344-4750-abbf-23d66d826bc4.png)

**Comment by wbecker** [comment]
I will try deleting images from document.body.paragraphs.items[x].inlinePictures instead as a work around.

**Comment by wbecker** [comment]
I have tried doing the same on a per paragraph basis:

```
  removeIssueButtonsFromDocument() {
    window.Word.run(
      async context => {

        const paragraphs = context.document.body.paragraphs;
        paragraphs.load("text");
        await context.sync();

        for (let paragraphIndex = 0; paragraphIndex < paragraphs.items.length; paragraphIndex += 1) {
          const paragraph = paragraphs.items[paragraphIndex];
          const pictures = paragraph.inlinePictures;
          pictures.load("altTextDescription");
          await context.sync();

          const picturesToDelete = pictures.items.filter(picture =>
            picture.altTextDescription.startsWith(MY_PREFIX)
          );

          if (picturesToDelete.length) {
            console.log("In paragraph", paragraphIndex, "deleting", picturesToDelete.length, "items");

            for (let index = 0; index < picturesToDelete.length; index += 1) {
              console.log("delete pic", index);
              const picture = picturesToDelete[index];
              picture.delete();
            }
            await context.sync();
          }
        }
      }
    );
  }
```

I note that it has the same problem (only a single paragraph is deleted at once) however, paragraph.inlinePictures is updated to reflect the actual amount of inlinePictures, so this function can be called multiple times until there are no inlinePictures left to remove. Not a perfect solution, but will work for the time being!

**Comment by JinghuiMS** [comment]
@wbecker 
Thanks for reporting this issue. It's definitely not right to have this error, but fortunately there is a workaround to unblock us. It has been put on our backlog<Bug#6539755> for internal track. We will keep track of this issue and let you know if there are any updates.

**Comment by YijunMS** [comment]
@wbecker 
Thanks for reporting this issue. I did some investigation and tried to reproduce it in **ScriptLab**.

For the two problems you mentioned:

1. In my tests, the length of `context.document.body.inlinePictures` **do** changed during the several calls. I guess the way we call the function `removeButtonsFromDocument` might be different. It would be helpful if you could provide more information about the calling way.
2. I do observe that it fails to delete all the pictures in a single call. I'm doing investigation on the root cause and trying to find a proper solution. As a **temporary workaround**, you could do the deletion in reverse order, and then it should delete all the pictures at one time. For example:
```
for (let index = picturesToDelete.length - 1; index >= 0; index -= 1) {
    console.log("delete pic", index);
    const picture = picturesToDelete[index];
    picture.delete();
    await context.sync();
}
```

We will let you know if there are any updates. Any additional information might be helpful.
